src/App.tsx

```diff
@@ -1,14 +1,80 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { FilterOption, SelectedId } from './types/variables';
+import { FilterCallback } from './types/functions';
+
+const doesTodoMeetQuery = (todo: Todo, query: string) => {
+  return todo.title.toLocaleLowerCase().includes(query.toLocaleLowerCase());
+};
+
+const getFilterOptionCallback = (
+  query: string,
+  option: FilterOption,
+): FilterCallback => {
+  switch (option) {
+    case FilterOption.All:
+      return (todo: Todo) => doesTodoMeetQuery(todo, query);
+    case FilterOption.Active:
+      return (todo: Todo) => !todo.completed && doesTodoMeetQuery(todo, query);
+    case FilterOption.Completed:
+      return (todo: Todo) => todo.completed && doesTodoMeetQuery(todo, query);
+    default:
+      throw new Error('Filter option is not valid!!!');
+  }
+};
+
+const getTodoById = (todos: Todo[], id: number): Todo | null => {
+  const foundTodo = todos?.find(todo => todo.id === id);
+
+  return foundTodo ? foundTodo : null;
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[] | null>(null);
+  const [selectedTodoId, setSelectedTodoId] = useState<SelectedId>(null);
+  const [filterOption, setFilterOption] = useState<FilterOption>(
+    FilterOption.All,
+  );
+  const [filterQuery, setFilterQuery] = useState('');
+
+  useEffect(() => {
+    getTodos().then(loadedTodos => setTodos(loadedTodos));
+  }, []);
+
+  const onTodoSelect = (id: SelectedId) => {
+    setSelectedTodoId(id);
+  };
+
+  const onFilterSelect = (option: FilterOption) => {
+    if (option !== filterOption) {
+      setFilterOption(option);
+    }
+  };
+
+  const onFilterChange = (query: string) => {
+    if (query !== filterQuery) {
+      setFilterQuery(query);
+    }
+  };
```

Comment:
[checklist](https://github.com/mate-academy/react_dynamic-list-of-todos/blob/master/checklist.md): follow [these](https://medium.com/javascript-in-plain-english/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364) naming conventions